### PR TITLE
Decouple direct connection from face selection to specific use.

### PR DESCRIPTION
**Pull request by rburema**

### CODE REVIEW

File: plugins/Tools/RotateTool/RotateTool.py [hunk 1]
```diff
@@ -56,9 +56,10 @@ def __init__(self):
         self._iterations = 0
         self._total_iterations = 0
         self._rotating = False
-        self.setExposedProperties("ToolHint", "RotationSnap", "RotationSnapAngle", "SelectFaceSupported")
+        self.setExposedProperties("ToolHint", "RotationSnap", "RotationSnapAngle", "SelectFaceSupported", "SelectFaceToLayFlatMode")
         self._saved_node_positions = []
 
+        self._selectFaceMode = False
```
Comment (nallath) [inline plugins/Tools/RotateTool/RotateTool.py]: Code style; should be select_face_mode

File: plugins/Tools/RotateTool/RotateTool.py [hunk 2]
```diff
@@ -264,6 +268,23 @@ def setRotationSnapAngle(self, angle):
             self._snap_angle = angle
             self.propertyChanged.emit()
 
+    ##  Wether the rotate tool is in 'Lay flat by face'-Mode.
+    #
+    #   \return (bool)
+    def getSelectFaceToLayFlatMode(self):
```
Comment (nallath) [inline plugins/Tools/RotateTool/RotateTool.py]: Missing mypy typing.

File: plugins/Tools/RotateTool/RotateTool.py [hunk 3]
```diff
@@ -264,6 +268,23 @@ def setRotationSnapAngle(self, angle):
...
+        if not Selection.getFaceSelectMode():
+            self._selectFaceMode = False  # .. but not the other way around!
+        return self._selectFaceMode
+
+    ##  Set the rotate tool to/from 'Lay flat by face'-Mode.
+    #
+    #   \param (bool)
+    def setSelectFaceToLayFlatMode(self, select):
```
Comment (nallath) [inline plugins/Tools/RotateTool/RotateTool.py]: Missing mypy typing

Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: Then we should do the typing of all the tools :-)

File: plugins/Tools/RotateTool/RotateTool.py (hunk 2 above)
Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: Then we should do the typing of all the tools :-)

File: plugins/Tools/RotateTool/RotateTool.py (hunk 1 above)
Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: Oops... will fix that right away.

File: plugins/Tools/RotateTool/RotateTool.py (hunk 2 above)
Comment (nallath) [reply plugins/Tools/RotateTool/RotateTool.py]: Oh, I figured this would fall under the boy scouting rule.

Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: Yes, you're right! Ok, will do. It just always looks so lopsided to me when _the entire rest of the file_ is not typed :'-D

Comment (nallath) [reply plugins/Tools/RotateTool/RotateTool.py]: Well, you could also do the rest ofcourse (and be a _really_ good boyscout)

Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: ... the entire class proved a bit much...

File: plugins/Tools/RotateTool/RotateTool.py (hunk 3 above)
Comment (Ghostkeeper) [reply plugins/Tools/RotateTool/RotateTool.py]: We've previously said that we'd add this typing boy-scout-style. When you edit something close by, add typing.

Comment (rburema) [reply plugins/Tools/RotateTool/RotateTool.py]: I think I remember* trying this, and then it gave more and more warnings (which would've tripped up the CI as well). So I added more typing, repeat. Eventually I had to conclude that it went way out of scope of the changes.

*) I'm not entirely sure it was for _this_ PR though.

**Pull request merged**